src/overlays/actors/ovl_En_Tp/z_en_tp.c

```diff
@@ -58,43 +79,699 @@ static ColliderJntSphInit D_80B22AB4 = {
     1,
     D_80B22A90,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Tp/func_80B20DE0.s")
 
+typedef enum {
+    /* 00 */ TAILPASARAN_DMGEFF_NONE, // No effect
+    /* 01 */ TAILPASARAN_DMGEFF_DEKUNUT,
+    /* 14 */ TAILPASARAN_DMGEFF_SHOCKING = 14, // Kills the Tailpasaran but shocks Player
+    /* 15 */ TAILPASARAN_DMGEFF_INSULATING     // Kills the Tailpasaran and does not shock Player
+} TailpasaranDamageEffect;
+
+static DamageTable D_80B22AC4 = {
+    /* Deku nut      */ DMG_ENTRY(0, TAILPASARAN_DMGEFF_DEKUNUT),
+    /* Deku stick    */ DMG_ENTRY(2, TAILPASARAN_DMGEFF_INSULATING),
+    /* Slingshot     */ DMG_ENTRY(0, TAILPASARAN_DMGEFF_NONE),
+    /* Explosive     */ DMG_ENTRY(0, TAILPASARAN_DMGEFF_NONE),
+    /* Boomerang     */ DMG_ENTRY(1, TAILPASARAN_DMGEFF_INSULATING),
+    /* Normal arrow  */ DMG_ENTRY(0, TAILPASARAN_DMGEFF_NONE),
+    /* Hammer swing  */ DMG_ENTRY(2, TAILPASARAN_DMGEFF_SHOCKING),
+    /* Hookshot      */ DMG_ENTRY(0, TAILPASARAN_DMGEFF_NONE),
+    /* Kokiri sword  */ DMG_ENTRY(1, TAILPASARAN_DMGEFF_SHOCKING),
+    /* Master sword  */ DMG_ENTRY(2, TAILPASARAN_DMGEFF_SHOCKING),
+    /* Giant's Knife */ DMG_ENTRY(4, TAILPASARAN_DMGEFF_SHOCKING),
+    /* Fire arrow    */ DMG_ENTRY(0, TAILPASARAN_DMGEFF_NONE),
+    /* Ice arrow     */ DMG_ENTRY(0, TAILPASARAN_DMGEFF_NONE),
+    /* Light arrow   */ DMG_ENTRY(0, TAILPASARAN_DMGEFF_NONE),
+    /* Unk arrow 1   */ DMG_ENTRY(0, TAILPASARAN_DMGEFF_NONE),
+    /* Unk arrow 2   */ DMG_ENTRY(0, TAILPASARAN_DMGEFF_NONE),
+    /* Unk arrow 3   */ DMG_ENTRY(0, TAILPASARAN_DMGEFF_NONE),
+    /* Fire magic    */ DMG_ENTRY(0, TAILPASARAN_DMGEFF_NONE),
+    /* Ice magic     */ DMG_ENTRY(0, TAILPASARAN_DMGEFF_NONE),
+    /* Light magic   */ DMG_ENTRY(0, TAILPASARAN_DMGEFF_NONE),
+    /* Shield        */ DMG_ENTRY(0, TAILPASARAN_DMGEFF_NONE),
+    /* Mirror Ray    */ DMG_ENTRY(0, TAILPASARAN_DMGEFF_NONE),
+    /* Kokiri spin   */ DMG_ENTRY(1, TAILPASARAN_DMGEFF_SHOCKING),
+    /* Giant spin    */ DMG_ENTRY(4, TAILPASARAN_DMGEFF_SHOCKING),
+    /* Master spin   */ DMG_ENTRY(2, TAILPASARAN_DMGEFF_SHOCKING),
+    /* Kokiri jump   */ DMG_ENTRY(2, TAILPASARAN_DMGEFF_SHOCKING),
+    /* Giant jump    */ DMG_ENTRY(8, TAILPASARAN_DMGEFF_SHOCKING),
+    /* Master jump   */ DMG_ENTRY(4, TAILPASARAN_DMGEFF_SHOCKING),
+    /* Unknown 1     */ DMG_ENTRY(0, TAILPASARAN_DMGEFF_NONE),
+    /* Unblockable   */ DMG_ENTRY(0, TAILPASARAN_DMGEFF_NONE),
+    /* Hammer jump   */ DMG_ENTRY(4, TAILPASARAN_DMGEFF_SHOCKING),
+    /* Unknown 2     */ DMG_ENTRY(0, TAILPASARAN_DMGEFF_NONE),
+};
+
+// sInitChain
+static InitChainEntry D_80B22AE4[] = {
+    ICHAIN_F32(targetArrowOffset, 10, ICHAIN_STOP),
+};
```

Comment:
```suggestion
static InitChainEntry sInitChain[] = {
    ICHAIN_F32(targetArrowOffset, 10, ICHAIN_STOP),
};
```
Also fix in the assembly file.

Reply:
Yeah I agree about updating the names for these static variables, since it should work fine now